File: main/obd_AKS/obd_script.py
## imports
import numpy as np
import main.obd.obd as ob
import imageio
import matplotlib.pyplot as plt
plt.style.use('dark_background')

## parameters
sf = np.array([50, 50])       # size of the PSF
maxiter = [50, 2]    # number of iterations for f and x
n = 40              # number of images
clipping = np.inf      # maximally acceptable pixel (for saturation correction)
srf = 1.0           # superresolution factor

# how are the filenames generated?
imagepath = 'obd/data/epsilon_lyrae/'

def y_fname(i):
    return imagepath+"{0:08d}".format(i)+'.png'

test = imageio.imread(y_fname(10))

# intially there is no x
x = np.array([])

for i in range(1,40):
  # load the next observed image
  fname = y_fname(i)
  print('Processing {}'.format(fname))
  y = imageio.imread(fname)
  y = y[:,20:132,2].astype(np.float64)   # use only first color channel

  ##### THE MAIN WORK HORSE #####
  x, f = ob.obd(x, y, sf, maxiter, clipping, srf)

  fig, ax = plt.subplots(1,3, figsize=(24., 8.))
  ax[0].imshow(y, origin='lower')
  ax[1].imshow(f, origin='lower')
  ax[2].imshow(x, origin='lower')
  plt.show()

File: obd_matlab/obd_script.m
% Script for
% Online Blind Deconvolution (OBD)
% Version 0.0, 10-Sep-2010
%
%    x     - true underlying image
%    y     - an observed image
%    f     - a point spread function (PSF)
%
% Copyright (C) 2010 by Michael Hirsch & Stefan Harmeling.

clear      % memory

% parameters
sf = [64, 64];       % size of the PSF
maxiter = [40, 1];   % number of iterations for f and x
n = 40;              % number of images
clipping = Inf;      % maximally acceptable pixel (for saturation correction)
srf = 1.0;           % superresolution factor

% how are the filenames generated?
imagepath = 'data/epsilon_lyrae';
y_fname = @(i) fullfile(imagepath, sprintf('%08d.png', i));

% intially there is no x
x = [];

% iterate over all images
for i = 1:n
  % load the next observed image
  fname = y_fname(i);
  fprintf('[%s.m] processing %s\n', mfilename, fname);
  y = imread(fname);
  y = double(y(:,21:132,1));   % use only first color channel

  %%%%% THE MAIN WORK HORSE %%%%%
  [x, f] = obd(x, y, sf, maxiter, clipping, srf);

%   if 1
%     % show intermediate output
%     clf
%     subplot(131), imagesc(y), title(sprintf('observed image y%d', i)); axis equal, axis tight
%     if exist('f', 'var')
%       subplot(132), imagesc(f), title(sprintf('estimated PSF f%d', i)); axis equal, axis tight
%     end
%     subplot(133), imagesc(x), title(sprintf('estimated image x%d', i)); axis equal, axis tight
%     %colormap gray
%     drawnow
%   end
end
fprintf('done!  the result is in variable "x", try e.g. "imagesc(x)"\n');
